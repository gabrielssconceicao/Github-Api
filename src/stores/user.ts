import { ref } from 'vue';
import { defineStore } from 'pinia';
import type { IUserResponse } from '@/types/user';
import { fetchUser } from '@/hooks/fetchUser';
export const useUserStore = defineStore('counter', () => {
  const githubUser = ref<IUserResponse>({} as IUserResponse);

  async function searchUser(userLogin: string) {
    const data = await fetchUser(userLogin);
    githubUser.value = data;
  }

  return { githubUser, searchUser };
});
