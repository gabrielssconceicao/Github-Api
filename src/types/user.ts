export interface IUser {
  name: null | string;
  login: string;
  avatar_url: string;
  followers: number;
  following: number;
  location: null | string;
  html_url: string;
  bio: null | string;
  public_repos: number;
  repos_url: string;
}

export interface IUserResponse {
  user: IUser;
  status: number;
}
