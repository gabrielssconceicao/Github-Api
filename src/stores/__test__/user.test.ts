import { describe, it, beforeEach, expect } from 'vitest';
import { setActivePinia, createPinia } from 'pinia';
import { useUserStore } from '../user';
import { userResponseError, userResponseSuccess } from '@/mock/__tests__/user';
describe('userStore', () => {
  beforeEach(() => {
    setActivePinia(createPinia());
  });

  it('should init with an empty user', () => {
    const userStore = useUserStore();
    expect(userStore.githubUser.status).toBe(undefined);
    expect(userStore.githubUser.user).toBe(undefined);
    expect(userStore).toMatchSnapshot();
  });

  it('should search user ', async () => {
    const userStore = useUserStore();
    await userStore.searchUser('resolveTest');
    expect(userStore.githubUser.status).toBe(userResponseSuccess.status);
    expect(userStore.githubUser.user).toEqual(userResponseSuccess.user);
    expect(userStore).toMatchSnapshot();
  });

  it('should search user error', async () => {
    const userStore = useUserStore();
    await userStore.searchUser('rejectTest');
    expect(userStore.githubUser.status).toBe(userResponseError.status);
    expect(userStore.githubUser.user).toEqual(userResponseError.user);
    expect(userStore).toMatchSnapshot();
  });
});
