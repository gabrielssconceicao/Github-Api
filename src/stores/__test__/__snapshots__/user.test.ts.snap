// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`userStore > should init with an empty user 1`] = `
{
  "$dispose": [Function],
  "$id": "counter",
  "$onAction": [Function],
  "$patch": [Function],
  "$reset": [Function],
  "$subscribe": [Function],
  "_customProperties": Set {},
  "_hmrPayload": {
    "actions": {
      "searchUser": [Function],
    },
    "getters": {},
    "hotState": RefImpl {
      "__v_isRef": true,
      "__v_isShallow": false,
      "_rawValue": {},
      "_value": {},
      "dep": undefined,
    },
    "state": [
      "githubUser",
    ],
  },
  "_hotUpdate": [Function],
  "_p": {
    "_a": null,
    "_e": EffectScope {
      "_active": true,
      "cleanups": [],
      "detached": true,
      "effects": [],
      "parent": undefined,
      "scopes": [
        EffectScope {
          "_active": true,
          "cleanups": [],
          "detached": false,
          "effects": [],
          "index": 0,
          "parent": [Circular],
        },
      ],
    },
    "_p": [],
    "_s": Map {
      "counter" => [Circular],
    },
    "install": [Function],
    "state": RefImpl {
      "__v_isRef": true,
      "__v_isShallow": false,
      "_rawValue": {
        "counter": {
          "githubUser": RefImpl {
            "__v_isRef": true,
            "__v_isShallow": false,
            "_rawValue": {},
            "_value": {},
            "dep": undefined,
          },
        },
      },
      "_value": {
        "counter": {
          "githubUser": {},
        },
      },
      "dep": undefined,
    },
    "use": [Function],
  },
  "githubUser": {},
  "searchUser": [Function],
}
`;

exports[`userStore > should search user  1`] = `
{
  "$dispose": [Function],
  "$id": "counter",
  "$onAction": [Function],
  "$patch": [Function],
  "$reset": [Function],
  "$subscribe": [Function],
  "_customProperties": Set {},
  "_hmrPayload": {
    "actions": {
      "searchUser": [Function],
    },
    "getters": {},
    "hotState": RefImpl {
      "__v_isRef": true,
      "__v_isShallow": false,
      "_rawValue": {},
      "_value": {},
      "dep": undefined,
    },
    "state": [
      "githubUser",
    ],
  },
  "_hotUpdate": [Function],
  "_p": {
    "_a": null,
    "_e": EffectScope {
      "_active": true,
      "cleanups": [],
      "detached": true,
      "effects": [],
      "parent": undefined,
      "scopes": [
        EffectScope {
          "_active": true,
          "cleanups": [],
          "detached": false,
          "effects": [],
          "index": 0,
          "parent": [Circular],
        },
      ],
    },
    "_p": [],
    "_s": Map {
      "counter" => [Circular],
    },
    "install": [Function],
    "state": RefImpl {
      "__v_isRef": true,
      "__v_isShallow": false,
      "_rawValue": {
        "counter": {
          "githubUser": RefImpl {
            "__v_isRef": true,
            "__v_isShallow": false,
            "_rawValue": {
              "status": 200,
              "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/10590076?v=4",
                "bio": "lorem ipsum dolor sit amet",
                "followers": 0,
                "following": 5,
                "html_url": "https://github.com/testeuser",
                "location": "Brazil",
                "login": "testeuser",
                "name": "Teste User",
                "public_repos": 5,
              },
            },
            "_value": {
              "status": 200,
              "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/10590076?v=4",
                "bio": "lorem ipsum dolor sit amet",
                "followers": 0,
                "following": 5,
                "html_url": "https://github.com/testeuser",
                "location": "Brazil",
                "login": "testeuser",
                "name": "Teste User",
                "public_repos": 5,
              },
            },
            "dep": undefined,
          },
        },
      },
      "_value": {
        "counter": {
          "githubUser": {
            "status": 200,
            "user": {
              "avatar_url": "https://avatars.githubusercontent.com/u/10590076?v=4",
              "bio": "lorem ipsum dolor sit amet",
              "followers": 0,
              "following": 5,
              "html_url": "https://github.com/testeuser",
              "location": "Brazil",
              "login": "testeuser",
              "name": "Teste User",
              "public_repos": 5,
            },
          },
        },
      },
      "dep": undefined,
    },
    "use": [Function],
  },
  "githubUser": {
    "status": 200,
    "user": {
      "avatar_url": "https://avatars.githubusercontent.com/u/10590076?v=4",
      "bio": "lorem ipsum dolor sit amet",
      "followers": 0,
      "following": 5,
      "html_url": "https://github.com/testeuser",
      "location": "Brazil",
      "login": "testeuser",
      "name": "Teste User",
      "public_repos": 5,
    },
  },
  "searchUser": [Function],
}
`;

exports[`userStore > should search user error 1`] = `
{
  "$dispose": [Function],
  "$id": "counter",
  "$onAction": [Function],
  "$patch": [Function],
  "$reset": [Function],
  "$subscribe": [Function],
  "_customProperties": Set {},
  "_hmrPayload": {
    "actions": {
      "searchUser": [Function],
    },
    "getters": {},
    "hotState": RefImpl {
      "__v_isRef": true,
      "__v_isShallow": false,
      "_rawValue": {},
      "_value": {},
      "dep": undefined,
    },
    "state": [
      "githubUser",
    ],
  },
  "_hotUpdate": [Function],
  "_p": {
    "_a": null,
    "_e": EffectScope {
      "_active": true,
      "cleanups": [],
      "detached": true,
      "effects": [],
      "parent": undefined,
      "scopes": [
        EffectScope {
          "_active": true,
          "cleanups": [],
          "detached": false,
          "effects": [],
          "index": 0,
          "parent": [Circular],
        },
      ],
    },
    "_p": [],
    "_s": Map {
      "counter" => [Circular],
    },
    "install": [Function],
    "state": RefImpl {
      "__v_isRef": true,
      "__v_isShallow": false,
      "_rawValue": {
        "counter": {
          "githubUser": RefImpl {
            "__v_isRef": true,
            "__v_isShallow": false,
            "_rawValue": {
              "status": 404,
              "user": {},
            },
            "_value": {
              "status": 404,
              "user": {},
            },
            "dep": undefined,
          },
        },
      },
      "_value": {
        "counter": {
          "githubUser": {
            "status": 404,
            "user": {},
          },
        },
      },
      "dep": undefined,
    },
    "use": [Function],
  },
  "githubUser": {
    "status": 404,
    "user": {},
  },
  "searchUser": [Function],
}
`;
